test/e2e/framework/fixture.go

```diff
@@ -282,47 +284,69 @@ func NewSyncerFixture(
 	t *testing.T,
 	resources sets.String,
 	upstream RunningServer,
-	orgClusterName logicalcluster.LogicalCluster,
 	wsClusterName logicalcluster.LogicalCluster,
 ) *SyncerFixture {
-	downstreamServer := NewFakeWorkloadServer(t, upstream, orgClusterName)
-
-	upstreamKcpClusterClient, err := kcpclientset.NewClusterForConfig(upstream.DefaultConfig(t))
+	// Write the upstream logical cluster config to disk for the workspace plugin
+	upstreamRawConfig, err := upstream.RawConfig()
 	require.NoError(t, err)
+	_, kubeconfigPath := writeLogicalClusterConfig(t, upstreamRawConfig, wsClusterName)
 
-	workloadCluster, err := CreateWorkloadCluster(t, upstream.Artifact, upstreamKcpClusterClient.Cluster(wsClusterName), downstreamServer)
-	require.NoError(t, err)
+	// TODO(marun) Configure this with a test arg to support syncer deployment to a pcluster
+	image := "foo"
+
+	// TODO(marun) Maybe support varying the workload cluster name to allow multiple syncers for a given workspace?
+	workloadClusterName := "pcluster-01"
 
-	upstreamConfig := upstream.DefaultConfig(t)
+	// Run the plugin command to enable the syncer and collect the resulting yaml
+	t.Logf("Configuring workspace %s for syncing", wsClusterName)
+	pluginArgs := []string{
+		"enable-syncer",
+		workloadClusterName,
+		"--syncer-image", image,
+	}
+	for _, resource := range resources.List() {
+		pluginArgs = append(pluginArgs, "--sync-resources", resource)
+	}
+	syncerYAML := RunWorkspacePlugin(t, kubeconfigPath, pluginArgs)
+
+	// Create a fake server from a workspace that is a peer to the current workspace
+	// TODO(marun) Only use a fake workload server if a pcluster is not configured
+	parentClusterName, ok := wsClusterName.Parent()
+	require.True(t, ok, "%s does not have a parent", wsClusterName)
+	downstreamServer := NewFakeWorkloadServer(t, upstream, parentClusterName)
```

Comment:
this confuses me.

Reply:
Can you elaborate here in a longer comment what you are doing with this fake workload servers? Is it just to test the cli plugin?

Reply:
It's to test the syncer without requiring a kind cluster. This isn't new code - we've been using it for some time now to test syncers - it's just being refactored. In #909 it becomes an optional path that is used when a pcluster (e.g. kind) isn't configured for the test run with `--pcluster-kubeconfig`.

Reply:
Updated in #909: https://github.com/kcp-dev/kcp/pull/909/files#diff-b2253283342ae684992dfd0124797a3d22622776caa42f2f18b2b021b05b914eR346